api/handlers/container/create.go

```diff
@@ -107,22 +108,65 @@ func (h *handler) create(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
-	// Annotations: TODO - available in nerdctl 2.0
 	// Annotations are passed in as a map of strings,
 	// but nerdctl expects an array of strings with format [annotations1=VALUE1, annotations2=VALUE2, ...].
-	// annotations := []string{}
-	// if req.HostConfig.Annotations != nil {
-	// 	for key, val := range req.HostConfig.Annotations {
-	// 		annotations = append(annotations, fmt.Sprintf("%s=%s", key, val))
-	// 	}
-	// }
+	annotations := []string{}
```

Comment:
nit: it would be nice to move all these conversions to individual functions with comprehensive unit tests.